### DISCUSSION TITLE
256gb emmc as a backup

### DISCUSSION THREAD

**Discussion by 3diamond** [Joshua-Riek/ubuntu-rockchip#534] [Q&A]
Hi Guys
I am trying to get the emmc module to be a backup of the main OS. Am i thinking in the correct way.. 
1. make a image of the OS on the nvme 
2. burn it on the emmc. 
3.
Maybe there is a mirror app that does that automatically?
There is a problem that Ubuntu sees the emmc as 17mb only instead of 256gb.
Many thanks

**Comment by bagong** [comment]
The 17mb block device is the SPI NOR, and internal flash memory that contains the bootloader. You should only right to it when you know exactly what you are doing (https://github.com/Joshua-Riek/ubuntu-rockchip/wiki/Orange-Pi-5-Plus), otherwise you might make it difficult to recover because the SoC can't boot so that you cannot easily correct the mistake. If you don't see your 256 GB emmc drive when issuing `lsblk` or looking in the Disks app, it was not recognized by the system. For me it shows up as SD-Card reader 250GB in Disks, and as `mmcblk0` 233GB in the list returned by `lsblk`. Maybe you didn't push the emmc properly into its socket. (As to backup, you might want to think about more efficient backup strategies than mirroring an entire drive. Often it's enough to back up user-data (incrementally). Ubuntu provides apps for that. If you're thinking of keeping the mirror as fallback device in case the nvme breaks (sort of a manually controlled raid1) you need to keep the boot sequence in mind which is not easily user changeable).

**Comment by 3diamond** [comment]
Thanks for answer. 
OK I get the 17mb block. good to know.
As for the emmc I think its dead. Brand new its was and official from orange pi. It just doesn't want to be recognized. Tried removing and pluging in. Checking if right way in case i did something stupid but to no avail. Maybe faulty. Unless there is a setting to switch on the emmc module.
I only want to mirror the operating system not whole drive as an alternative if one should breakdown. But this can't be done yet till emmc is sorted.
Thank you

**Reply by bagong** [reply]
Well, hopefully someone more knowledgeable than me is willing to chime in. I wonder if rewriting the bootloader to the SPI NOR as described in the link above might help. I think it defines what happens in the early boot stage, and that might very well concern whether the emmc is ignored further on or not.

**Comment by 3diamond** [comment]
Hi 
I just got the replacement emmc from "orange" its still not being detected. Maybe it needs to be setup in dev tree(maybe port is not activated)? which file would it be?
